pkg/frontend/admin_openshiftcluster_etcdcertificaterenew_test.go

```diff
@@ -0,0 +1,508 @@
+						},
+					},
+				},
+			},
+			etcdoperator: &operatorv1.Etcd{
+				Status: operatorv1.EtcdStatus{
+					StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+						OperatorStatus: operatorv1.OperatorStatus{
+							Conditions: []operatorv1.OperatorCondition{
+								{
+									Type:   "EtcdCertSignerControllerDegraded",
+									Status: operatorv1.ConditionFalse,
+								},
+							},
+						},
+					},
+				},
+			},
+			etcdCO: &configv1.ClusterOperator{
+				Status: configv1.ClusterOperatorStatus{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						{
+							Type:   configv1.OperatorDegraded,
+							Status: configv1.ConditionTrue,
+						},
+					},
+				},
+			},
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").MaxTimes(1).
+					Return(encodeClusterVersion(t, tt.version), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "etcd.operator.openshift.io", "", "cluster").MaxTimes(1).
+					Return(encodeEtcdOperatorController(t, tt.etcdoperator), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterOperator.config.openshift.io", "", "etcd").
+					Return(encodeEtcdOperator(t, tt.etcdCO), nil)
+			},
+			wantStatusCode: http.StatusInternalServerError,
+			wantError:      "500: InternalServerError: : Etcd Operator is not in expected state, quiting.",
+		},
+		{
+			name:       "validate if etcd certificates are not near expiry",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			version: &configv1.ClusterVersion{
+				Status: configv1.ClusterVersionStatus{
+					History: []configv1.UpdateHistory{
+						{
+							State:   configv1.CompletedUpdate,
+							Version: "4.8.11",
+						},
+					},
+				},
+			},
+			etcdoperator: &operatorv1.Etcd{
+				Status: operatorv1.EtcdStatus{
+					StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+						OperatorStatus: operatorv1.OperatorStatus{
+							Conditions: []operatorv1.OperatorCondition{
+								{
+									Type:   "EtcdCertSignerControllerDegraded",
+									Status: operatorv1.ConditionFalse,
+								},
+							},
+						},
+					},
+				},
+			},
+			etcdCO: &configv1.ClusterOperator{
+				Status: configv1.ClusterOperatorStatus{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						{
+							Type:   configv1.OperatorDegraded,
+							Status: configv1.ConditionFalse,
+						},
+					},
+				},
+			},
+			notBefore: time.Now(),
+			notAfter:  time.Now().AddDate(3, 0, 0),
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").MaxTimes(1).
+					Return(encodeClusterVersion(t, tt.version), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "etcd.operator.openshift.io", "", "cluster").AnyTimes().
```

Comment:
I think you want to check that it is called at least once, I think that AnyTimes() can mean 0